### ISSUE TITLE
0.8.0 fails to build with the latest go: cannot use "quic-go doesn't build on Go 1.18 yet." (untyped string constant) as int value in variable declaration

### ISSUE THREAD

**Issue by yurivict** [ipfs/ipget#104]
```
github.com/lucas-clemente/quic-go/internal/qtls
# github.com/lucas-clemente/quic-go/internal/qtls
vendor/github.com/lucas-clemente/quic-go/internal/qtls/go118.go:6:13: cannot use "quic-go doesn't build on Go 1.18 yet." (untyped string constant) as int value in variable declaration
```

OS: FreeBSD 13

**Comment by aschmahmann** [comment]
https://github.com/lucas-clemente/quic-go/wiki/quic-go-and-Go-versions

Basically, this can get resolved once go-ipfs updates to a version of QUIC that supports 1.18. See the link for more details.

**Comment by yurivict** [comment]
Any chance to resolve this? Currently ipget fails to build with the latest version of Go.

**Comment by yurivict** [comment]
go-ipfs is at 0.12.2, but ipget still uses 1.11

**Comment by lidel** [comment]
latest go-ipfs 0.13-rc1 builds fine with golag 1.18, so if someone wants to work on this:
-  `ipget` needs to be updated at least to that version of go-ipfs, and also bump go-libp2p to 0.19 or later

**Comment by yurivict** [comment]
0.8.1 still has this problem.

I think I should just remove the port because the project is broken for months.

**Comment by dmgk** [comment]
Polite ping. go-ipfs currently is at v0.13.0, and builds fine with go1.18.

**Comment by Stebalien** [comment]
This will be fixed by #110. Unfortunately, that's blocked on https://github.com/ipfs/go-ipfs-http-client/pull/160.

**Comment by lidel** [comment]
Should be fixed in v0.9.x: https://dist.ipfs.io/ipget
